projects/core/src/user/facade/user-consent.service.ts

```diff
@@ -119,7 +127,12 @@ export class UserConsentService {
    */
   getConsent(templateId: string): Observable<Consent | undefined> {
     return this.authService.isUserLoggedIn().pipe(
-      filter(Boolean),
+      distinctUntilChanged(),
+      // to avoid redundant calls when userId is still ANONYMOUS
+      withLatestFrom(this.userIdService.getUserId()),
+      filter(
+        ([loggedIn, userId]) => loggedIn && userId === OCC_USER_ID_CURRENT
```

Comment:
Sure, I agree, but I was wondering whether we want to add now a change to such a crucial part of an application (actually parts because the ASM feature can be activated independently and then AsmAuthService is in use, so `isUserLoggedIn in this service has to be extended as well)
Proposal for `AsmAuthService`:
```ts
  isUserLoggedIn(): Observable<boolean> {
    return combineLatest([
      this.authStorageService.getToken(),
      this.authStorageService.getTokenTarget(),
      this.userIdService.isEmulated(),
      this.userIdService.getUserId(),
    ]).pipe(
      map(([token, tokenTarget, isEmulated, userId]) => {
        const isCurrentUser =
          tokenTarget === TokenTarget.User && userId === OCC_USER_ID_CURRENT;
        const isEmulatedUser =
          tokenTarget === TokenTarget.CSAgent && isEmulated;
        return (
          Boolean(token?.access_token) && (isCurrentUser || isEmulatedUser)
        );
      }),
      distinctUntilChanged()
    );
  }
```
Proposal for `AuthService`:
```ts
  isUserLoggedIn(): Observable<boolean> {
    return combineLatest([
      this.authStorageService.getToken(),
      this.userIdService.getUserId(),
    ]).pipe(
      map(([userToken, userId]) => {
        console.log('userID', userId);
        return Boolean(userToken?.access_token);
      }),
      distinctUntilChanged()
    );
  }
```

Reply:
As discussed offline: Let's now fix the problem only in the symptom for 6.0. And in 6.x let's fix it in the roots.